Kryo prints a lot of class loading failures

**Describe the bug**
During investigating https://github.com/zeebe-io/zeebe/issues/5127 I found the following messages in the log

```
I 2020-08-10T09:00:43.034371344Z 1177:47  WARN: [kryo] Unable to load class ��� �������������e with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.035851662Z 1177:47  WARN: [kryo] Unable to load class ��� �������������f with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.036390170Z 1177:47  WARN: [kryo] Unable to load class ��� �������������g with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.037425588Z 1177:47  WARN: [kryo] Unable to load class ��� �������������h with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.038462281Z 1177:47  WARN: [kryo] Unable to load class ��� �������������i with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.039447158Z 1177:47  WARN: [kryo] Unable to load class ��� �������������j with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.040062484Z 1177:47  WARN: [kryo] Unable to load class ��� �������������k with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.040568978Z 1177:47  WARN: [kryo] Unable to load class ���HG�������������l with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.041273564Z 1177:47  WARN: [kryo] Unable to load class ���px7 with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.042711536Z 1177:47  WARN: [kryo] Unable to load class ���7hb7�������������s with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.043388448Z 1177:47  WARN: [kryo] Unable to load class ���7hb7�������������t with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.043618996Z 1177:47  WARN: [kryo] Unable to load class ���7hb7�������������u with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.044744082Z 1177:47  WARN: [kryo] Unable to load class ���7hb7�������������v with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.045427433Z 1177:47  WARN: [kryo] Unable to load class ���7hb7�������������w with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.046031821Z 1177:47  WARN: [kryo] Unable to load class ���68 with kryo's ClassLoader. Retrying with current..
 
```
![output](https://user-images.githubusercontent.com/2758593/89766859-1c04a600-daf9-11ea-901b-67447e5f47d8.png)



![p2-last-role-change](https://user-images.githubusercontent.com/2758593/89768600-18265300-dafc-11ea-8653-35572e638e0c.png)

If we take a look when the last role change happen for partition 2, then we can see that previous Leader Broker 2 has the similar issue and after Broker 1 transition to leader he prints the log's endless.


```
I 2020-08-10T01:03:25.842166364Z zeebe-chaos-zeebe-2 702:09  WARN: [kryo] Unable to load class ���HG�������������#$ with kryo's ClassLoader. Retrying with current..
  zeebe-chaos-zeebe-2
I 2020-08-10T01:03:25.842565011Z zeebe-chaos-zeebe-2 702:09  WARN: [kryo] Unable to load class ���HG�������������$$ with kryo's ClassLoader. Retrying with current..
  zeebe-chaos-zeebe-2
I 2020-08-10T01:03:25.842835305Z zeebe-chaos-zeebe-2 702:09  WARN: [kryo] Unable to load class ���HG�������������%$ with kryo's ClassLoader. Retrying with current..
  zeebe-chaos-zeebe-2
I 2020-08-10T01:03:25.843160564Z zeebe-chaos-zeebe-2 702:09  WARN: [kryo] Unable to load class ���HG�������������&$ with kryo's ClassLoader. Retrying with current..
  zeebe-chaos-zeebe-2
I 2020-08-10T01:03:25.843434758Z zeebe-chaos-zeebe-2 702:09  WARN: [kryo] Unable to load class ���HG�������������'$ with kryo's ClassLoader. Retrying with current..
  zeebe-chaos-zeebe-2
I 2020-08-10T01:03:25.843863183Z zeebe-chaos-zeebe-2 702:09  WARN: [kryo] Unable to load class ���HG�������������($ with kryo's ClassLoader. Retrying with current..
  zeebe-chaos-zeebe-2
I 2020-08-10T01:03:25.844272852Z zeebe-chaos-zeebe-2 702:09  WARN: [kryo] Unable to load class ����������������)$ with kryo's ClassLoader. Retrying with current..
  zeebe-chaos-zeebe-2
I 2020-08-10T01:03:26.278650Z zeebe-chaos-zeebe-0 RaftServer{raft-partition-partition-1}{role=FOLLOWER} - No heartbeat from null in the last PT2.856S (calculated from last 2856 ms), sending poll requests  zeebe-chaos-zeebe-0
I 2020-08-10T01:03:27.569731306Z zeebe-chaos-zeebe-2 java.lang.OutOfMemoryError: Java heap space
  zeebe-chaos-zeebe-2
I 2020-08-10T01:03:27.569757849Z zeebe-chaos-zeebe-2 Dumping heap to /usr/local/zeebe/data/java_pid6.hprof ...
  zeebe-chaos-zeebe-2
I 2020-08-10T01:03:27.569764732Z zeebe-chaos-zeebe-2 Unable to create /usr/local/zeebe/data/java_pid6.hprof: File exists
  zeebe-chaos-zeebe-2
I 2020-08-10T01:03:27.569769677Z zeebe-chaos-zeebe-2 Terminating due to java.lang.OutOfMemoryError: Java heap space
  zeebe-chaos-zeebe-2
W 2020-08-10T01:03:27.966019Z zeebe-chaos-zeebe-0 RaftServer{raft-partition-partition-1}{role=FOLLOWER} - Poll request to 2 failed: java.net.ConnectException  zeebe-chaos-zeebe-0
I 2020-08-10T01:03:27.968927571Z zeebe-chaos-zeebe-1 {"severity":"WARNING","logging.googleapis.com/sourceLocation":{"function":"failAttempt","file":"AbstractAppender.java","line":229},"message":"RaftServer{raft-partition-partition-1} - AppendRequest{term=249, leader=1, prevLogIndex=700415776, prevLogTerm=249, entries=19, commitIndex=700415763} to 2 failed: java.util.concurrent.CompletionException: java.net.ConnectException","serviceContext":{"service":"zeebe-broker","version":"zeebe-chaos"},"context":{"threadId":76,"threadPriority":5,"loggerName":"io.atomix.raft.roles.LeaderAppender","threadName":"raft-server-1-raft-partition-partition-1"},"exception":"java.util.concurrent.CompletionException: java.net.ConnectException\n\tat java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source) ~[?:?]\n\tat io.atomix.cluster.messaging.impl.NettyMessagingService.lambda$executeOnPooledConnection$17(NettyMessagingService.java:509) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]\n\tat com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30) ~[guava-29.0-jre.jar:?]\n\tat io.atomix.cluster.messaging.impl.NettyMessagingService.lambda$executeOnPooledConnection$18(NettyMessagingService.java:509) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]\n\tat java.util.concurrent.CompletableFuture.uniWhenComplete(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source) ~[?:?]\n\tat io.atomix.cluster.messaging.impl.AbstractClientConnection$Callback.completeExceptionally(AbstractClientConnection.java:186) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]\n\tat io.atomix.cluster.messaging.impl.AbstractClientConnection.close(AbstractClientConnection.java:123) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]\n\tat io.atomix.cluster.messaging.impl.NettyMessagingService.lambda$getOrCreateClientConnection$30(NettyMessagingService.java:578) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]\n\tat io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:577) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:551) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]{"severity":"WARNING","logging.googleapis.com/sourceLocation":{"function":"failAttempt","file":"AbstractAppender.java","line":229},"message":"RaftServer{raft-partition-partition-3} - AppendRequest{term=232, leader=1, prevLogIndex=692875289, prevLogTerm=232, entries=5, commitIndex=692889465} to 2 failed: java.util.concurrent.CompletionException: java.net.ConnectException","serviceContext":{"service":"zeebe-broker","version":"zeebe-chaos"},"context":{"threadId":79,"threadPriority":5,"loggerName":"io.atomix.raft.roles.LeaderAppender","threadName":"raft-server-1-raft-partition-partition-3"},"exception":"java.util.concurrent.CompletionException: java.net.ConnectException\n\tat java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source) ~[?:?]\n\tat io.atomix.cluster.messaging.impl.NettyMessagingService.lambda$executeOnPooledConnection$17(NettyMessagingService.java:509) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]\n\tat com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30) ~[guava-29.0-jre.jar:?]\n\tat io.atomix.cluster.messaging.impl.NettyMessagingService.lambda$executeOnPooledConnection$18(NettyMessagingService.java:509) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]\n\tat java.util.concurrent.CompletableFuture.uniWhenComplete(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source) ~[?:?]\n\tat io.atomix.cluster.messaging.impl.AbstractClientConnection$Callback.completeExceptionally(AbstractClientConnection.java:186) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]\n\tat io.atomix.cluster.messaging.impl.AbstractClientConnection.close(AbstractClientConnection.java:123) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]\n\tat io.atomix.cluster.messaging.impl.NettyMessagingService.lambda$getOrCreateClientConnection$30(NettyMessagingService.java:578) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]\n\tat io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:577) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:551) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:490) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:615) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:604) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:104) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannel$CloseFuture.setClosed(AbstractChannel.java:1158) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannel$AbstractUnsafe.doClose0(AbstractChannel.java:760) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:736) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:607) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.shutdownInput(AbstractEpollChannel.java:484) ~[netty-transport-native-epoll-4.1.51.Final-linux-x86_64.jar:4.1.51.Final]\n\tat io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:815) ~[netty-transport-native-epoll-4.1.51.Final-linux-x86_64.jar:4.1.51.Final]\n\tat io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:475) ~[netty-transport-native-epoll-4.1.51.Final-linux-x86_64.jar:4.1.51.Final]\n\tat io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378) ~[netty-transport-native-epoll-4.1.51.Final-linux-x86_64.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat java.lang.Thread.run(Unknown Source) [?:?]\nCaused by: java.net.ConnectException\n\t... 20 more\n","timestampSeconds":1597021407,"timestampNanos":965514000,"logger":"io.atomix.raft.roles.LeaderAppender","thread":"raft-server-1-raft-partition-partition-1"}
  zeebe-chaos-zeebe-1
I 2020-08-10T01:03:27.969000371Z zeebe-chaos-zeebe-1 \n\tat io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:490) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:615) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:604) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:104) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannel$CloseFuture.setClosed(AbstractChannel.java:1158) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannel$AbstractUnsafe.doClose0(AbstractChannel.java:760) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:736) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:607) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.DefaultChannelPipeline$HeadContext.close(DefaultChannelPipeline.java:1352) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeClose(AbstractChannelHandlerContext.java:622) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannelHandlerContext.close(AbstractChannelHandlerContext.java:606) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannelHandlerContext.close(AbstractChannelHandlerContext.java:472) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.atomix.cluster.messaging.impl.AbstractMessageEncoder.exceptionCaught(AbstractMessageEncoder.java:152) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:302) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:281) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannelHandlerContext.fireExceptionCaught(AbstractChannelHandlerContext.java:273) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.DefaultChannelPipeline$HeadContext.exceptionCaught(DefaultChannelPipeline.java:1377) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:302) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:281) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.DefaultChannelPipeline.fireExceptionCaught(DefaultChannelPipeline.java:907) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.handleReadException(AbstractEpollStreamChannel.java:728) ~[netty-transport-native-epoll-4.1.51.Final-linux-x86_64.jar:4.1.51.Final]\n\tat io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:818) ~[netty-transport-native-epoll-4.1.51.Final-linux-x86_64.jar:4.1.51.Final]\n\tat io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:475) ~[netty-transport-native-epoll-4.1.51.Final-linux-x86_64.jar:4.1.51.Final]\n\tat io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378) ~[netty-transport-native-epoll-4.1.51.Final-linux-x86_64.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat java.lang.Thread.run(Unknown Source) [?:?]\nCaused by: java.net.ConnectException\n\t... 32 more\n","timestampSeconds":1597021407,"timestampNanos":965507000,"logger":"io.atomix.raft.roles.LeaderAppender","thread":"raft-server-1-raft-partition-partition-3"}
  zeebe-chaos-zeebe-1
I 2020-08-10T01:03:27.970810772Z zeebe-chaos-zeebe-1 {"severity":"WARNING","logging.googleapis.com/sourceLocation":{"function":"failAttempt","file":"AbstractAppender.java","line":229},"message":"RaftServer{raft-partition-partition-3} - AppendRequest{term=232, leader=1, prevLogIndex=692875294, prevLogTerm=232, entries=14, commitIndex=692889475} to 2 failed: java.util.concurrent.CompletionException: java.net.ConnectException","serviceContext":{"service":"zeebe-broker","version":"zeebe-chaos"},"context":{"threadId":79,"threadPriority":5,"loggerName":"io.atomix.raft.roles.LeaderAppender","threadName":"raft-server-1-raft-partition-partition-3"},"exception":"java.util.concurrent.CompletionException: java.net.ConnectException\n\tat java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source) ~[?:?]\n\tat io.atomix.cluster.messaging.impl.NettyMessagingService.lambda$executeOnPooledConnection$17(NettyMessagingService.java:509) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]\n\tat com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30) ~[guava-29.0-jre.jar:?]\n\tat io.atomix.cluster.messaging.impl.NettyMessagingService.lambda$executeOnPooledConnection$18(NettyMessagingService.java:509) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]\n\tat java.util.concurrent.CompletableFuture.uniWhenComplete(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source) ~[?:?]\n\tat io.atomix.cluster.messaging.impl.AbstractClientConnection$Callback.completeExceptionally(AbstractClientConnection.java:186) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]\n\tat io.atomix.cluster.messaging.impl.AbstractClientConnection.close(AbstractClientConnection.java:123) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]\n\tat io.atomix.cluster.messaging.impl.NettyMessagingService.lambda$getOrCreateClientConnection$30(NettyMessagingService.java:578) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]\n\tat io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:577) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:551) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]{"severity":"WARNING","logging.googleapis.com/sourceLocation":{"function":"failAttempt","file":"AbstractAppender.java","line":229},"message":"RaftServer{raft-partition-partition-1} - AppendRequest{term=249, leader=1, prevLogIndex=700415763, prevLogTerm=249, entries=13, commitIndex=700415761} to 2 failed: java.util.concurrent.CompletionException: java.net.ConnectException","serviceContext":{"service":"zeebe-broker","version":"zeebe-chaos"},"context":{"threadId":76,"threadPriority":5,"loggerName":"io.atomix.raft.roles.LeaderAppender","threadName":"raft-server-1-raft-partition-partition-1"},"exception":"java.util.concurrent.CompletionException: java.net.ConnectException\n\tat java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source) ~[?:?]\n\tat io.atomix.cluster.messaging.impl.NettyMessagingService.lambda$executeOnPooledConnection$17(NettyMessagingService.java:509) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]\n\tat com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30) ~[guava-29.0-jre.jar:?]\n\tat io.atomix.cluster.messaging.impl.NettyMessagingService.lambda$executeOnPooledConnection$18(NettyMessagingService.java:509) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]\n\tat java.util.concurrent.CompletableFuture.uniWhenComplete(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]\n\tat java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source) ~[?:?]\n\tat io.atomix.cluster.messaging.impl.AbstractClientConnection$Callback.completeExceptionally(AbstractClientConnection.java:186) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]\n\tat io.atomix.cluster.messaging.impl.AbstractClientConnection.close(AbstractClientConnection.java:123) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]\n\tat io.atomix.cluster.messaging.impl.NettyMessagingService.lambda$getOrCreateClientConnection$30(NettyMessagingService.java:578) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]\n\tat io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:577) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:551) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:490) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:615) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:604) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:104) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannel$CloseFuture.setClosed(AbstractChannel.java:1158) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannel$AbstractUnsafe.doClose0(AbstractChannel.java:760) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:736) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:607) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.DefaultChannelPipeline$HeadContext.close(DefaultChannelPipeline.java:1352) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeClose(AbstractChannelHandlerContext.java:622) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannelHandlerContext.close(AbstractChannelHandlerContext.java:606) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannelHandlerContext.close(AbstractChannelHandlerContext.java:472) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.atomix.cluster.messaging.impl.AbstractMessageEncoder.exceptionCaught(AbstractMessageEncoder.java:152) ~[atomix-cluster-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:302) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:281) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannelHandlerContext.fireExceptionCaught(AbstractChannelHandlerContext.java:273) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.DefaultChannelPipeline$HeadContext.exceptionCaught(DefaultChannelPipeline.java:1377) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:302) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:281) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.DefaultChannelPipeline.fireExceptionCaught(DefaultChannelPipeline.java:907) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:490) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:615) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:604) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:104) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannel$CloseFuture.setClosed(AbstractChannel.java:1158) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannel$AbstractUnsafe.doClose0(AbstractChannel.java:760) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:736) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:607) ~[netty-transport-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.shutdownInput(AbstractEpollChannel.java:484) ~[netty-transport-native-epoll-4.1.51.Final-linux-x86_64.jar:4.1.51.Final]\n\tat io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:815) ~[netty-transport-native-epoll-4.1.51.Final-linux-x86_64.jar:4.1.51.Final]\n\tat io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:475) ~[netty-transport-native-epoll-4.1.51.Final-linux-x86_64.jar:4.1.51.Final]\n\tat io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378) ~[netty-transport-native-epoll-4.1.51.Final-linux-x86_64.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat java.lang.Thread.run(Unknown Source) [?:?]\nCaused by: java.net.ConnectException\n\t... 20 more\n","timestampSeconds":1597021407,"timestampNanos":969302000,"logger":"io.atomix.raft.roles.LeaderAppender","thread":"raft-server-1-raft-partition-partition-1"}
  zeebe-chaos-zeebe-1
I 2020-08-10T01:03:27.970884400Z zeebe-chaos-zeebe-1 \n\tat io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.handleReadException(AbstractEpollStreamChannel.java:728) ~[netty-transport-native-epoll-4.1.51.Final-linux-x86_64.jar:4.1.51.Final]\n\tat io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:818) ~[netty-transport-native-epoll-4.1.51.Final-linux-x86_64.jar:4.1.51.Final]\n\tat io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:475) ~[netty-transport-native-epoll-4.1.51.Final-linux-x86_64.jar:4.1.51.Final]\n\tat io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378) ~[netty-transport-native-epoll-4.1.51.Final-linux-x86_64.jar:4.1.51.Final]\n\tat io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.51.Final.jar:4.1.51.Final]\n\tat java.lang.Thread.run(Unknown Source) [?:?]\nCaused by: java.net.ConnectException\n\t... 32 more\n","timestampSeconds":1597021407,"timestampNanos":969302000,"logger":"io.atomix.raft.roles.LeaderAppender","thread":"raft-server-1-raft-partition-partition-3"}
  zeebe-chaos-zeebe-1
W 2020-08-10T01:03:27.976480Z zeebe-chaos-zeebe-1 RaftServer{raft-partition-partition-3} - AppendRequest{term=232, leader=1, prevLogIndex=692875308, prevLogTerm=232, entries=0, commitIndex=692891977} to 2 failed: java.util.concurrent.CompletionException: io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: zeebe-chaos-zeebe-2.zeebe-chaos-zeebe.zeebe-chaos.svc.cluster.local/10.0.18.8:26502  zeebe-chaos-zeebe-1
W 2020-08-10T01:03:28.009161Z zeebe-chaos-zeebe-1 RaftServer{raft-partition-partition-1} - AppendRequest{term=249, leader=1, prevLogIndex=700415795, prevLogTerm=249, entries=0, commitIndex=700415763} to 2 failed: java.util.concurrent.CompletionException: io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: zeebe-chaos-zeebe-2.zeebe-chaos-zeebe.zeebe-chaos.svc.cluster.local/10.0.18.8:26502  zeebe-chaos-zeebe-1
I 2020-08-10T01:03:28.939362Z zeebe-chaos-zeebe-1 RaftServer{raft-partition-partition-2}{role=FOLLOWER} - No heartbeat from DefaultRaftMember{id=2, type=ACTIVE, updated=2020-07-31T12:29:53.224Z} in the last PT3.226S (calculated from last 3226 ms), sending poll requests  zeebe-chaos-zeebe-1
W 2020-08-10T01:03:28.943904Z zeebe-chaos-zeebe-1 RaftServer{raft-partition-partition-2}{role=FOLLOWER} - Poll request to 2 failed: io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: zeebe-chaos-zeebe-2.zeebe-chaos-zeebe.zeebe-chaos.svc.cluster.local/10.0.18.8:26502  zeebe-chaos-zeebe-1
I 2020-08-10T01:03:28.943943Z zeebe-chaos-zeebe-0 RaftServer{raft-partition-partition-2}{role=FOLLOWER} - Accepted PollRequest{term=104, candidate=1, lastLogIndex=15641265, lastLogTerm=104}: candidate's log is up-to-date  zeebe-chaos-zeebe-0
I 2020-08-10T01:03:28.946227Z zeebe-chaos-zeebe-1 RaftServer{raft-partition-partition-2} - Transitioning to CANDIDATE  zeebe-chaos-zeebe-1
I 2020-08-10T01:03:28.948097Z zeebe-chaos-zeebe-1 RaftServer{raft-partition-partition-2}{role=CANDIDATE} - Starting election  zeebe-chaos-zeebe-1
W 2020-08-10T01:03:28.955867Z zeebe-chaos-zeebe-1 RaftServer{raft-partition-partition-2}{role=CANDIDATE} - io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: zeebe-chaos-zeebe-2.zeebe-chaos-zeebe.zeebe-chaos.svc.cluster.local/10.0.18.8:26502  zeebe-chaos-zeebe-1
I 2020-08-10T01:03:28.962552Z zeebe-chaos-zeebe-0 RaftServer{raft-partition-partition-2}{role=FOLLOWER} - Accepted VoteRequest{term=105, candidate=1, lastLogIndex=15641265, lastLogTerm=104}: candidate's log is up-to-date  zeebe-chaos-zeebe-0
I 2020-08-10T01:03:28.965755Z zeebe-chaos-zeebe-1 RaftServer{raft-partition-partition-2} - Transitioning to LEADER  zeebe-chaos-zeebe-1
I 2020-08-10T01:03:28.966590Z zeebe-chaos-zeebe-1 RaftServer{raft-partition-partition-2} - Found leader 1  zeebe-chaos-zeebe-1
I 2020-08-10T01:03:29.008702783Z zeebe-chaos-zeebe-1 700:33  WARN: [kryo] Unable to load class ���px7 with kryo's ClassLoader. Retrying with current..
  zeebe-chaos-zeebe-1
I 2020-08-10T01:03:29.011748520Z zeebe-chaos-zeebe-1 700:33  WARN: [kryo] Unable to load class ���7hb7�������������;a with kryo's ClassLoader. Retrying with current..
  zeebe-chaos-zeebe-1
I 2020-08-10T01:03:29.012152321Z zeebe-chaos-zeebe-1 700:33  WARN: [kryo] Unable to load class ���7hb7�������������<a with kryo's ClassLoader. Retrying with current..
  zeebe-chaos-zeebe-1
I 2020-08-10T01:03:29.012615576Z zeebe-chaos-zeebe-1 700:33  WARN: [kryo] Unable to load class ���7hb7�������������=a with kryo's ClassLoader. Retrying with current..
  zeebe-chaos-zeebe-1
I 2020-08-10T01:03:29.012887777Z zeebe-chaos-zeebe-1 700:33  WARN: [kryo] Unable to load class ����H with kryo's ClassLoader. Retrying with current..
  zeebe-chaos-zeebe-1
I 2020-08-10T01:03:29.013353341Z zeebe-chaos-zeebe-1 700:33  WARN: [kryo] Unable to load class ��� with kryo's ClassLoader. Retrying with current..
  zeebe-chaos-zeebe-1
I 2020-08-10T01:03:29.013639800Z zeebe-chaos-zeebe-1 700:33  WARN: [kryo] Unable to load class ����������������Aa with kryo's ClassLoader. Retrying with current..
  zeebe-chaos-zeebe-1
I 2020-08-10T01:03:29.013935268Z zeebe-chaos-zeebe-1 700:33  WARN: [kryo] Unable to load class ��� �������������Ba with kryo's ClassLoader. Retrying with current..
  zeebe-chaos-zeebe-1
I 2020-08-10T01:03:29.014185585Z zeebe-chaos-zeebe-1 700:33  WARN: [kryo] Unable to load class ��� �������������Ca with kryo's ClassLoader. Retrying with current..
  zeebe-chaos-zeebe-1
I 2020-08-10T01:03:29.014463265Z zeebe-chaos-zeebe-1 700:33  WARN: [kryo] Unable to load class ��� �������������Da with kryo's ClassLoader. Retrying with current..
  zeebe-chaos-zeebe-1
```

Did a short discussion with @MiguelPires about the logs we have seen here https://github.com/zeebe-io/zeebe/issues/5127#issuecomment-671242640 . 

The problem with our benchmark is that we use `camunda:SNAPSHOT` and `pullPolicy: Always` which means it will deploy a newer snapshot if the broker was restarted or node was  preempted. This means that we have the last changes in our running benchmarks which have been merged later. The benchmark was setup ~10 days again the PR https://github.com/zeebe-io/zeebe/pulls?q=is%3Apr+is%3Aclosed+author%3AMiguelPires was merged 7 days ago.

If we use some stackdriver magic then we can see that this log happens after the PR was merged.

![log-kryo](https://user-images.githubusercontent.com/2758593/89780315-7f9acd80-db11-11ea-8df5-2a8ee278951c.png)

@MiguelPires  is able to reproduce it with a new Benchmark and will take a closer look. He think this is not a real issue and means it is expected if we read/receive old messages. If we take a look at https://github.com/zeebe-io/zeebe/issues/5127#issuecomment-671290156, then we see that we have a lot of old data in our log.


**To Reproduce**

Start a benchmark.

**Expected behavior**
Not that many error logs.

**Environment:**
 - OS: k8
 - Zeebe Version: snapshot
 - Configuration: elastic

--- a/atomix/utils/pom.xml
+++ b/atomix/utils/pom.xml
@@ -14,7 +14,9 @@
   ~ See the License for the specific language governing permissions and
   ~ limitations under the License.
   -->
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <artifactId>atomix-utils</artifactId>
 
   <dependencies>
@@ -78,6 +80,10 @@
       <groupId>org.assertj</groupId>
       <artifactId>assertj-core</artifactId>
       <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>com.esotericsoftware</groupId>
+      <artifactId>minlog</artifactId>
     </dependency>
   </dependencies>
 

--- a/atomix/utils/src/main/java/io/atomix/utils/serializer/NamespaceImpl.java
+++ b/atomix/utils/src/main/java/io/atomix/utils/serializer/NamespaceImpl.java
@@ -29,14 +29,13 @@ import com.esotericsoftware.kryo.pool.KryoFactory;
 import com.esotericsoftware.kryo.pool.KryoPool;
 import com.esotericsoftware.kryo.serializers.CompatibleFieldSerializer;
 import com.esotericsoftware.kryo.serializers.VersionFieldSerializer;
+import com.esotericsoftware.minlog.Log;
 import com.google.common.base.MoreObjects;
 import com.google.common.collect.ImmutableList;
-import io.atomix.utils.config.ConfigurationException;
 import java.io.ByteArrayInputStream;
 import java.nio.ByteBuffer;
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
 import org.apache.commons.lang3.tuple.Pair;
@@ -78,13 +77,8 @@ public class NamespaceImpl implements Namespace, KryoFactory, KryoPool {
   private final boolean registrationRequired;
   private final String friendlyName;
 
-  public NamespaceImpl(final NamespaceConfig config) {
-    this(
-        buildRegistrationBlocks(config),
-        Thread.currentThread().getContextClassLoader(),
-        config.isRegistrationRequired(),
-        config.isCompatible(),
-        config.getName());
+  static {
+    Log.NONE();
   }
 
   /**
@@ -95,7 +89,7 @@ public class NamespaceImpl implements Namespace, KryoFactory, KryoPool {
    * @param compatible whether compatible serialization is enabled
    * @param friendlyName friendly name for the namespace
    */
-  private NamespaceImpl(
+  public NamespaceImpl(
       final List<RegistrationBlock> registeredTypes,
       final ClassLoader classLoader,
       final boolean registrationRequired,
@@ -106,33 +100,6 @@ public class NamespaceImpl implements Namespace, KryoFactory, KryoPool {
     this.classLoader = classLoader;
     this.compatible = compatible;
     this.friendlyName = checkNotNull(friendlyName);
-  }
-
-  @SuppressWarnings("unchecked")
-  private static List<RegistrationBlock> buildRegistrationBlocks(final NamespaceConfig config) {
-    final List<Pair<Class<?>[], Serializer<?>>> types = new ArrayList<>();
-    final List<RegistrationBlock> blocks = new ArrayList<>();
-    blocks.addAll(Namespaces.BASIC.registeredBlocks);
-    for (final NamespaceTypeConfig type : config.getTypes()) {
-      try {
-        if (type.getId() == null) {
-          types.add(
-              Pair.of(
-                  new Class[] {type.getType()},
-                  type.getSerializer() != null ? type.getSerializer().newInstance() : null));
-        } else {
-          blocks.add(
-              new RegistrationBlock(
-                  type.getId(),
-                  Collections.singletonList(
-                      Pair.of(new Class[] {type.getType()}, type.getSerializer().newInstance()))));
-        }
-      } catch (final InstantiationException | IllegalAccessException e) {
-        throw new ConfigurationException("Failed to instantiate serializer from configuration", e);
-      }
-    }
-    blocks.add(new RegistrationBlock(FLOATING_ID, types));
-    return blocks;
   }
 
   /**

--- a/parent/pom.xml
+++ b/parent/pom.xml
@@ -1,4 +1,6 @@
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
 
   <modelVersion>4.0.0</modelVersion>
   <name>Zeebe Parent</name>
@@ -58,6 +60,7 @@
     <version.java-grpc-prometheus>0.3.0</version.java-grpc-prometheus>
     <version.junit>4.13</version.junit>
     <version.log4j>2.13.3</version.log4j>
+    <version.minlog>1.3.1</version.minlog>
     <version.mockito>3.4.6</version.mockito>
     <version.model>7.7.0</version.model>
     <version.msgpack>0.8.20</version.msgpack>
@@ -665,6 +668,11 @@
         <version>${version.failsafe}</version>
       </dependency>
 
+      <dependency>
+        <groupId>com.esotericsoftware</groupId>
+        <artifactId>minlog</artifactId>
+        <version>${version.minlog}</version>
+      </dependency>
     </dependencies>
   </dependencyManagement>
 
@@ -968,7 +976,7 @@
                     </goals>
                   </pluginExecutionFilter>
                   <action>
-                    <ignore />
+                    <ignore/>
                   </action>
                 </pluginExecution>
               </pluginExecutions>
@@ -1061,18 +1069,18 @@
               </goals>
               <configuration>
                 <rules>
-                  <dependencyConvergence />
+                  <dependencyConvergence/>
                 </rules>
               </configuration>
             </execution>
-             <execution>
+            <execution>
               <id>enforce-unique-dependencies</id>
               <goals>
                 <goal>enforce</goal>
               </goals>
               <configuration>
                 <rules>
-                  <banDuplicatePomDependencyVersions />
+                  <banDuplicatePomDependencyVersions/>
                 </rules>
               </configuration>
             </execution>

